### DISCUSSION TITLE
[question] How to scope buffers to tab pages?

### DISCUSSION THREAD

**Discussion by midrare** [noib3/cokeline.nvim#108] [Ideas]
Say if I open a buffer when a certain tab page is open, I want that buffer to appear in `cokeline` only when that tab page is open. Under any other tab page, that buffer is hidden.

Is this possible in `cokeline`, and do you have any general pointers for how to best implement this? (Alternatively, would this be a good new feature to be built into `cokeline`?)

**edit:** Also, scoping buffers to windows?

**Comment by yyy33** [comment]
I think this is a cool feature

**Comment by midrare** [comment]
I found a workaround: use [bufferline.nvim](https://github.com/akinsho/bufferline.nvim) with [scope.nvim](https://github.com/tiagovla/scope.nvim). I'd still prefer a tabline-native solution though.

**Comment by willothy** [comment]
Absolutely planning on adding this feature! 

I'd like to be able to scope/group buffers to tabs and collapse the tab groups for more free space.

**Comment by willothy** [comment]
Moved this to here for discussion purposes, feature updates will be posted in #35.

**Comment by midrare** [comment]
I see that the feature went through, but after reading the docs it's not obvious to me how to implement scoping buffers to tabs/windows. Any tips?

**Reply by willothy** [reply]
I use [`scope.nvim`](https://github.com/tiagovla/scope.nvim), and I'd suggest others to do so as well. It works very well with cokeline, no extra configuration needed. I may still implement tab scoping here at some point, but given that `scope` does it very well it's not a high priority for me. My next goal for cokeline is to make all component updates event driven / asynchronous - after that, I may begin adding more features again.

I'll make an issue for this and try to get to it as soon as I can.
